src/audiomixerboard.cpp

```diff
@@ -1210,92 +1210,98 @@ void CAudioMixerBoard::ApplyNewConClientList ( CVector<CChannelInfo>& vecChanInf
 
         // search for channels with are already present and preserve their gain
         // setting, for all other channels reset gain
-        for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
+
+        // get all channels which are in use/not in use. We use the array index of vecChanInfo
+        // if the fader is in use, else INVALID_INDEX to specify it is not in use
+        int iFaderNumber[MAX_NUM_CHANNELS];
+
+        for ( int iChanID = 0; iChanID < MAX_NUM_CHANNELS; iChanID++ )
+        {
+            iFaderNumber[iChanID] = INVALID_INDEX;
+        }
+
+        for ( int iFader = 0; iFader < iNumConnectedClients; iFader++ )
         {
-            bool bFaderIsUsed = false;
+            iFaderNumber[vecChanInfo[iFader].iChanID] = iFader;
+        }
+
+        for ( int iChanID = 0; iChanID < MAX_NUM_CHANNELS; iChanID++ )
+        {
+            if ( iFaderNumber[iChanID] == INVALID_INDEX )
+            {
+                // fader is not used --> hide it
```

Comment:
1232/1233 could be rewritten to flow better.  "Hide it."  "Oh wait!! Before that, just one thing - store the levels.  Oh!! Only of certain conditions are met."

So first we store the levels (I don't think the "if some..." is needed - that's the concern of the called method - though maybe the method name needs improving at some point), then we hide the fader.  The comment should reflect that.

However..!  I don't even think the comment is adding to the code here.  "Store...; ...Hide()" is fairly self-explanatory (as the vector name is good).